draft-ietf-rats-corim.md

```diff
@@ -957,24 +957,30 @@ If a `flags-map` value is not specified, then the operational mode is unknown.
 
 The following describes each member of the `flags-map`:
 
-* `configured` (index 0): The measured environment is fully configured for
-  normal operation if the flag is true.
+* `configured` (index 0): If the flag is true, the measured environment is fully configured for normal operation.
 
-* `secure` (index 1): The measured environment's configurable security settings
-  are fully enabled if the flag is true.
+* `secure` (index 1): If the flag is true, the measured environment's configurable
+security settings are fully enabled.
 
-* `recovery` (index 2): The measured environment is NOT in a recovery state if
-  the flag is true.
+* `recovery` (index 2): If the flag is true, the measured environment is NOT in recovery
```

Comment:
Just wondering how come we reversed the logic of this flag.

Ideally if `recovery` is true then measured environment should be in the recovery mode isn't it?

Reply:
> Just wondering how come we reversed the logic of this flag.

Originally, the first 3 flags were defined by TCG Attestation Architecture spec. They were described in ASN.1 as a bit array. They wanted the bit array to be zero if no security significant states were present. Interestingly, the debug flag in comid didn't preserve the reverse logic. Subsequent additions to the flags haven't preserved this convention. The most recent definition of flags in the asn.1 structure includes a mask and a max-size marker making it less important for the bit array to be zero for no security relevant concerns.

It seems odd that recovery has negative logic. Does it make sense to change this? What is the impact?

Reply:
`++massive_cognitive_dissonance;`

Not sure who's responsible for fixing this though.

We could drop it from this PR and make an issue to solve this with upstream, or put it in anyway but still have an explicit, inline warning flag and associated issue.

Happy either way.

Reply:
Either way, we need to raise this issue. @nedmsmith : I am OK, if you track this as a defect to fix it downstream in all appropriate documents...

All I do not want this `massive cognitive dissonance` to percolate to exist forever and be pain to readers/interpreters.

Reply:
Fixed in latest push.